boot/zephyr/include/flash_map/flash_map.h

```diff
@@ -43,14 +43,64 @@ extern "C" {
  */
 #include <inttypes.h>
 
+/**
+ * @brief Structure describing an area of memory-mapped flash.
```

Comment:
That makes sense; I'll reword this. 

I think that also implies that we need a separate flash_device_base()-like routine which can return error in case the device isn't memory mapped. Keeping the current code, but putting real addresses into fa_off, wouldn't let callers tell the two situations apart, and hilarity could ensue.